### PR TITLE
adding from_stack and append to LabelMapAxis

### PR DESCRIPTION
**Pull request by MRegeard**
This pull request is related to pull request #4409. 

I added `from_stack` and `append` method to `LabelMapAxis` in order to be able to use `LabelMapAxis` as an axis for `Map.from_stack`.

### CODE REVIEW

**Comment by adonath** [comment]
Thanks @MRegeard, this looks very good! Please add a small test show that the functionality works as expected.

**Review by QRemy — DISMISSED** [review]
Thanks @MRegeard, looks good !

File: gammapy/maps/axes.py [hunk 1]
```diff
@@ -3147,3 +3147,50 @@ def slice(self, idx):
...
+        axis_stacked = axes[0]
+
+        for ax in axes[1:]:
+            axis_stacked = axis_stacked.append(ax)
+
+        return axis_stacked
+
+    def append(self, axis):
+        """Append another label map axis to this label map axis.
+        Name and node_type must agree between the axes. labels must be unique.
```
Comment (registerrier) [inline gammapy/maps/axes.py:3175]:
```suggestion
        """Append another label map axis to this label map axis.
        
        Names must agree between the axes. Labels must be unique.
```

File: gammapy/maps/axes.py [hunk 2]
```diff
@@ -3147,3 +3147,50 @@ def slice(self, idx):
...
+
+        Returns
+        -------
+        axis : `LabelMapAxis`
+            Appended axis
+        """
+        if self.node_type != axis.node_type:
+            raise ValueError(
+                f"Node type must agree, got {self.node_type} and {axis.node_type}"
+            )
```
Comment (registerrier) [inline gammapy/maps/axes.py:3190]: This should not be necessary as `LabelMapAxis` does not define `node_type`. You might check that the other axis is a `LabelMapAxis` instead.

**Review by registerrier — APPROVED** [review]
Thanks @MRegeard . This looks good. See the inline comments.

File: gammapy/maps/axes.py (hunk 2 above)
Comment (MRegeard) [reply gammapy/maps/axes.py]: Done

File: gammapy/maps/axes.py (hunk 1 above)
Comment (MRegeard) [reply gammapy/maps/axes.py]: Done

File: gammapy/maps/axes.py [hunk 3]
```diff
@@ -3147,3 +3147,52 @@ def slice(self, idx):
...
+            Axis to append.
+
+        Returns
+        -------
+        axis : `LabelMapAxis`
+            Appended axis
+        """
+        if not isinstance(axis, LabelMapAxis):
+            raise TypeError(
+                f"axis must be an instance of {self.__name__}, got {axis.__name__}"
```
Comment (registerrier) [inline gammapy/maps/axes.py:3190]:
```suggestion
                f"axis must be an instance of LabelMapAxis, got {axis.__class__.__name__} instead."
```

**Review by registerrier — APPROVED** [review]
Thanks @MRegeard

File: gammapy/maps/axes.py (hunk 3 above)
Comment (MRegeard) [reply gammapy/maps/axes.py]: Done

**Review by registerrier — APPROVED** [review]
Thanks @MRegeard . No more comments.

**Pull request merged**